nucliadb/src/nucliadb/ingest/service/writer.py

```diff
@@ -101,12 +97,40 @@ async def finalize(self): ...
     async def NewKnowledgeBox(  # type: ignore
         self, request: KnowledgeBoxNew, context=None
     ) -> NewKnowledgeBoxResponse:
+        if is_onprem_nucliadb():
+            logger.error(
+                "Sorry, this endpoint is only available for hosted. Onprem must use the REST API"
+            )
+            return NewKnowledgeBoxResponse(
+                status=KnowledgeBoxResponseStatus.ERROR,
+            )
+
+        # Hosted KBs are created through backend endpoints. We assume learning
+        # configuration has been already created for it and we are given the
+        # model metadata in the request
+
+        kbid = request.forceuuid or KnowledgeBoxORM.new_unique_kbid()
+        release_channel = request.release_channel
+        semantic_model = parse_model_metadata_from_request(request)
+
         try:
-            kbid = await self.create_kb(request)
-            logger.info("KB created successfully", extra={"kbid": kbid})
+            async with self.driver.transaction() as txn:
+                kbid, failed = await KnowledgeBoxORM.create(
```

Comment:
now that we are at it, can we simply raise exceptions here and not return the `failed`?

Reply:
Sure!